sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -196,6 +196,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val NO_UNESCAPED_SQL_STRING = buildConf("spark.sql.noUnescapedStringLiteral")
```

Comment:
Double negatives are not encouraged in conf naming. This sounds the first parser conf. 

How about `spark.sql.parser.escapeStringLiterals`?